### PR TITLE
INSPECT-5789 soft id validator

### PR DESCRIPTION
**Pull request by laurence-smith**
Adds a soft validation option for the id validator.
This is to allow gathering data with soft validation before we enforce stricter validations.
Failed soft validations are inserted into logs via `fmt.Println` with the same message an `fmt.Error` would produce.
Upgrade go version to 1.18

### CODE REVIEW

File: s12/protobuf/proto/validator.proto [hunk 1]
```diff
@@ -214,6 +214,11 @@ message IdRules {
   // the beginning of the string.
   // @TODO: implement if we need this, e.g. for user_UUID, org_UUID or use a dedicated validation type for this
   // optional string prefix = 5;
+
+  // Only apply soft validation. Validation errors for this validator are logged, not returned.
+  // Be aware that due to multiples validators applying, and order of precedence
+  // this will not prevent errors from other validators being returned.
+  optional bool soft_validation = 5 [default = false];
```
Comment (sc-peter) [inline s12/protobuf/proto/validator.proto:221]: Thanks @laurence-smith, what do you think about the suggestion to rename this to `monitor_mode` or `report_only` or `log_only`?
It's a good idea to test validation for existing fields first so we could add this to other validators as well later. Can you please also update the README.md?

Comment (MickStanciu) [reply s12/protobuf/proto/validator.proto:221]: log_only is a sensitive solution

File: protobuf/protoc-gen-govalidator/plugin/plugin.go [hunk 2]
```diff
@@ -627,7 +627,12 @@ func genIdValidator(g *protogen.GeneratedFile, f *protogen.Field, varName string
 
 	// Check if any of the validations passed
 	g.P("if !isValidId {")
-	genErrorString(g, varName, string(f.Desc.Name()), errMsg)
+	if rules.GetLogOnly() {
+		printErrorString(g, varName, string(f.Desc.Name()), errMsg)
```
Comment (sc-peter) [inline protobuf/protoc-gen-govalidator/plugin/plugin.go:631]: Suggest to prefix the log line with `[log-only] ` so you can distinguish between requests that have actually been rejected.

Comment (MickStanciu) [reply protobuf/protoc-gen-govalidator/plugin/plugin.go:631]: done

**Review by sc-peter — APPROVED** [review]

**Pull request merged**